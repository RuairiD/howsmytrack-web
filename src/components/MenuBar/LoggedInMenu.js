import React from "react";

import { Badge, Menu, Icon } from "antd";
import { A, Span } from "lemon-reset";

import UserDetails from "./UserDetails";

const LoggedInMenu = ({ onMenuClick, username, rating, notifications }) => (
    <Menu
        theme="light"
        onClick={onMenuClick}
        style={{
            marginBottom: "auto",
            borderRight: 0,
        }}
    >
        <UserDetails
            username={username}
            rating={rating}
        />
        <Menu.Item key="newRequest">
            <Icon type="plus" />
            <Span>New Request</Span>
        </Menu.Item>
        <Menu.Item key="feedbackGroups">
            <A href="/groups">
                <Icon type="team" />
                <Span>Your Groups</Span>
                <Badge
                    count={notifications}
                    style={{ marginLeft: "0.5em" }}
                />
            </A>
        </Menu.Item>
        <Menu.Item key="faq">
            <A href="/faq">
                <Icon type="question-circle" />
                <Span>FAQ</Span>
            </A>
        </Menu.Item>
        <Menu.Item key="settings">
            <A href="/settings">
                <Icon type="setting" />
                <Span>Settings</Span>
            </A>
        </Menu.Item>
        <Menu.Item key="logout">
            <Icon type="user-delete" />
            <Span>Sign Out</Span>
        </Menu.Item>
    </Menu>
);

export default LoggedInMenu;
