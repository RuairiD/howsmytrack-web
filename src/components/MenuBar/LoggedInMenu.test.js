import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoggedInMenu from "./LoggedInMenu";

describe("LoggedInMenu", () => {
    it("renders menu for logged in users with New Request, Your Groups, FAQ, Settings and Sign Out options", () => {
        const wrapper = shallow(
            <LoggedInMenu
                onMenuClick={() => {}}
                username="username"
                rating={4.321}
                notifications={2}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
