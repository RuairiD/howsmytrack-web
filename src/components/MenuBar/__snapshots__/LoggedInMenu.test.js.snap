// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInMenu renders menu for logged in users with New Request, Your Groups, FAQ, Settings and Sign Out options 1`] = `
<Menu
  onClick={[Function]}
  style={
    Object {
      "borderRight": 0,
      "marginBottom": "auto",
    }
  }
  theme="light"
>
  <UserDetails
    rating={4.321}
    username="username"
  />
  <MenuItem
    key="newRequest"
  >
    <Icon
      type="plus"
    />
    <Span
      className=""
    >
      New Request
    </Span>
  </MenuItem>
  <MenuItem
    key="feedbackGroups"
  >
    <A
      className=""
      href="/groups"
    >
      <Icon
        type="team"
      />
      <Span
        className=""
      >
        Your Groups
      </Span>
      <Badge
        count={2}
        dot={false}
        overflowCount={99}
        showZero={false}
        style={
          Object {
            "marginLeft": "0.5em",
          }
        }
      />
    </A>
  </MenuItem>
  <MenuItem
    key="faq"
  >
    <A
      className=""
      href="/faq"
    >
      <Icon
        type="question-circle"
      />
      <Span
        className=""
      >
        FAQ
      </Span>
    </A>
  </MenuItem>
  <MenuItem
    key="settings"
  >
    <A
      className=""
      href="/settings"
    >
      <Icon
        type="setting"
      />
      <Span
        className=""
      >
        Settings
      </Span>
    </A>
  </MenuItem>
  <MenuItem
    key="logout"
  >
    <Icon
      type="user-delete"
    />
    <Span
      className=""
    >
      Sign Out
    </Span>
  </MenuItem>
</Menu>
`;
